pkgs/package-overrides.nix

```diff
@@ -181,13 +199,17 @@ in
           ourPatches =
             lib.optionals (lib.versionOlder prev.php.version "8.0") [
               # Header path defaults to FHS location, preventing the configure script from detecting errno support.
-              # TODO: re-add when PHP 7 is removed from Nixpkgs.
-              # ./patches/iconv-header-path.patch
+              ./patches/iconv-header-path.patch
             ];
         in
         ourPatches ++ upstreamPatches;
     });
 
+    json = prev.mkExtension {
+      name = "json";
+      enable = lib.versionOlder prev.php.version "8.0";
```

Comment:
`enable` does not work here (that is only used for filtering in https://github.com/NixOS/nixpkgs/blob/d9065632716ae90db984dfbe133ee01b5495209d/pkgs/top-level/php-packages.nix#L581). You need to use `if-then-else` and return `null` when not enabled.